packages/opds-web-client/src/components/context/StoreContext.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react'
+import {Provider} from 'react-redux'
+import * as Redux from 'redux'
+import { State } from "../../state";
+import AuthPlugin from "../../AuthPlugin";
+import buildStore from "../../store";
+import {PathForContext} from './PathForContext';
+import BasicAuthPlugin from "../../BasicAuthPlugin";
+
+type OPDSStoreProps = {
+  children: React.ReactElement;
+  initialState?: State;
+  authPlugins?: AuthPlugin[];
+}
+/**
+ * Builds the redux store and makes it available in context.
+ * takes in the pathFor context. Will be used by OPDSCatalog
+ * as well as circulation-patron-web.
+ */
+export default class OPDSStore extends React.Component<OPDSStoreProps>{
+  static contextType = PathForContext;
+  context: React.ContextType<typeof PathForContext>;
+
+  store: Redux.Store<State>;
+
+  constructor(props, context) {
+    super(props);
+    this.store = buildStore(this.props.initialState || undefined, this.props.authPlugins || [BasicAuthPlugin], context.pathFor);
+  }
+
+  render() {
+    return (
+      <Provider store={this.store}>
+        {this.props.children}
+      </Provider>
```

Comment:
Why doesn't this use a similar pattern like `RouterContext.tsx` where there is a `createContext` variable to use in the render function? Is it because the `store` attribute is explicitly available in this provider?

Reply:
Yes exactly, it is because the `context` variable is already created by `redux`, we are just importing it, rendering the provider, passing in a value to the provider, and also passing the context down via the legacy API as well. Basically, redux already created it for us